### DISCUSSION TITLE
Permissions to modify only specific field on specific objet.

### DISCUSSION THREAD

**Discussion by dannymongrain1** [netbox-community/netbox#15448] [Q&A]
Hello. I need help to setup limitations to a specific permission.  I need to grant a 'modify' Action on a specific Object Type but I don't want all field to be modifiable, only a specific one.

Example.  on object-type 'ipam | IP address', i need Actions 'Modify' but the only field that can be modify is 'description'.  I don't want the user which match this permission to modify the role, the vrf, the status, etc.

Is that possible?

**Comment by candlerb** [comment]
No, there are no field-level permissions, and [Custom Validators](https://docs.netbox.dev/en/stable/customization/custom-validation/) don't get access to the before+after attributes or the identity of the user making the change.

If you want to do this in the Netbox UI, I think you need make the object read-only, and then write a [Custom Script](https://docs.netbox.dev/en/stable/customization/custom-scripts/) which lets the user update the field of interest.  Your script will then do its own validation as to whether the user is allowed to update this field or not.

**Reply by dannymongrain1** [reply]
thank you for the answer, we'll try a custom script
